packages/nodes-base/nodes/Grist/Grist.node.ts

```diff
@@ -0,0 +1,247 @@
+	parseSortProperties,
+} from './GenericFunctions';
+
+import {
+	operationFields,
+} from './OperationDescription';
+
+import {
+	GristColumns,
+	GristCreateRowPayload,
+	GristDefinedData,
+	GristGetAllOptions,
+	GristUpdateRowPayload,
+} from './types';
+
+export class Grist implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Grist',
+		name: 'grist',
+		icon: 'file:grist.svg',
+		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+		group: ['input'],
+		version: 1,
+		description: 'Consume the Grist API',
+		defaults: {
+			name: 'Grist',
+			color: '#394650',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		credentials: [
+			{
+				name: 'gristApi',
+				required: true,
+			},
+		],
+		properties: operationFields,
+	};
+
+	methods = {
+		loadOptions: {
+			async getTableColumns(this: ILoadOptionsFunctions) {
+				const docId = this.getNodeParameter('docId', 0) as string;
+				const tableId = this.getNodeParameter('tableId', 0) as string;
+				const endpoint = `/docs/${docId}/tables/${tableId}/columns`;
+
+				const { columns } = await gristApiRequest.call(this, 'GET', endpoint) as GristColumns;
+				return columns.map(({ id }) => ({ name: id, value: id }));
+			},
+		},
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		let responseData;
+		const returnData: IDataObject[] = [];
+
+		const operation = this.getNodeParameter('operation', 0) as string;
+
+		for (let i = 0; i < items.length; i++) {
+
+			try {
+
+				if (operation === 'create') {
+
+					// ----------------------------------
+					//             create
+					// ----------------------------------
+
+					// https://support.getgrist.com/api/#tag/records/paths/~1docs~1{docId}~1tables~1{tableId}~1records/post
+
+					const body = { records: [] } as GristCreateRowPayload;
+
+					const dataToSend = this.getNodeParameter('dataToSend', 0) as 'defineBelow' | 'autoMapInputData';
+
+					if (dataToSend === 'autoMapInputData') {
+						const incomingKeys = Object.keys(items[i].json);
+						const rawInputsToIgnore = this.getNodeParameter('inputsToIgnore', i) as string;
+						const inputDataToIgnore = rawInputsToIgnore.split(',').map(c => c.trim());
+
+						for (const key of incomingKeys) {
+							if (inputDataToIgnore.includes(key)) continue;
+							body.records[i] = { fields: { [key]: items[i].json[key] } };
+						}
+
+					} else if (dataToSend === 'defineBelow') {
+						const { properties } = this.getNodeParameter('fieldsToSend', i, []) as {
+							properties: GristDefinedData;
+						};
+
+						if (!properties.length) {
+							throw new NodeOperationError(this.getNode(), 'Please enter data to send');
+						}
+
+						body.records = [
+							{ fields: parseFieldsToSend(properties) },
+						];
+					}
+
+					const docId = this.getNodeParameter('docId', i) as string;
+					const tableId = this.getNodeParameter('tableId', i) as string;
```

Comment:
Should these be allowed to vary for different input items? `getTableColumns` doesn't which makes fields have fixed options. Are users allowed to enter an expression in `fieldId` which doesn't match those options?